### DISCUSSION TITLE
Next.js Link Component Not Updating Dynamic Pages on Subsequent Navigation

### DISCUSSION THREAD

**Discussion by meet-hypeteq** [vercel/next.js#59760] [Help] [answered]
### Summary

href link is given using next/link.
But for dynamic pages, when right click and select open in new tab then page is not load and no api call send.

**Comment by ghost** [comment]
The issue you're experiencing with the Next.js `Link` component not updating dynamic pages on subsequent navigation is a known problem in the Next.js community. This behavior happens because Next.js `Link` component performs client-side navigation and seems to serve a cached state of the previously visited component [Source 1](https://stackoverflow.com/questions/74884114/nextjs-reload-page-with-link-component). The client never calls back to the server, and thus the server component never re-renders on the second navigation [Source 1](https://stackoverflow.com/questions/74884114/nextjs-reload-page-with-link-component).

A workaround for this issue is to remove the `Link` component and use a regular anchor (`<a>`) tag instead [Source 0](https://github.com/vercel/next.js/issues/49300). However, this approach removes any loading animations that might be present when using the `Link` component.

Alternatively, you can use the `useRouter` hook from 'next/router' to programmatically navigate between pages. This hook allows you to push or replace the current route. However, as reported in the community, there might be some issues with stale data when using `router.replace` or `router.push` with a page that has `export const dynamic = 'force-dynamic'` [Source 0](https://github.com/vercel/next.js/issues/49300).

```jsx
import { useRouter } from 'next/router'

export const MyComponent = () => {
  const router = useRouter()

  const navigateToDynamicPage = () => {
    router.push('/dynamic')
  }

  return (
    <button onClick={navigateToDynamicPage}>
      Go to dynamic page
    </button>
  )
}
```

Another solution mentioned in the community involves adding a unique key to the component that changes with each navigation [Source 2](https://github.com/vercel/next.js/discussions/22512). This forces React to remount the component, triggering a re-render.

```jsx
import { useRouter } from 'next/router'

export default function App({ Component, pageProps }) {
  const router = useRouter()

  return <Component {...pageProps} key={router.asPath} />
}
```

Lastly, you can use a check inside a `useEffect` hook to update a state value if the `query.id` changes. This can cause the component to re-render and can fix the issue with page data not updating correctly [Source 2](https://github.com/vercel/next.js/discussions/22512).


```
import { useRouter } from 'next/router'
import { useEffect, useState } from 'react'

export const MyComponent = () => {
  const router = useRouter()
  const [id, setId] = useState(router.query.id)

  useEffect(() => {
    if (router.query.id !== id) {
      setId(router.query.id)
    }
  }, [router.query.id])

  // Your component logic here
}
```


Remember that these are workarounds and the behavior might change in future versions of Next.js. It's always a good idea to keep track of the issue on the official Next.js GitHub page to get updates and possible fixes.

**Reply by meet-hypeteq** [reply]
Thanks @ashishvala15 for the reply. After next JS version 13, need to use next/navigation instead of next/router. so this solution works for older version.

**Comment by meet-hypeteq** [comment, marked as answer]
It might be due to hydration error or size of first load js during build is red.
Below solution worked for us,

In app.jsx file,

<Suspense fallback={<></>}>
      <Body>{children}</Body>
  </Suspense>
